api/main.py

```
@@ -0,0 +1,752 @@
+    )
+    
+    # Upload da imagem do cardápio para o Supabase Storage
+    menu_image_url = None
+    if menu_image:
+        try:
+            menu_image_url = await services.upload_image_to_supabase(menu_image)
+        except HTTPException as e:
+            raise e # Re-raise HTTPException from service layer
+        except Exception as e:
+            logger.error(f"Erro ao fazer upload da imagem do cardápio para o Supabase: {e}", exc_info=True)
+            raise HTTPException(status_code=500, detail="Erro interno ao fazer upload da imagem do cardápio.")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)